### PR TITLE
feat: make react-query a peerDependency + update to react-query 5+

### PR DESCRIPTION
**Pull request by LuizAsFight**

### CODE REVIEW

File: packages/react/src/providers/FuelUIProvider.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ export function FuelUIProvider({
 }: FuelUIProviderProps) {
   const [theme, setTheme] = useState(initialTheme || 'light');
   const { fuel } = useFuel();
-  const { isLoading: isConnecting, isError, connect } = useConnect();
+  const { isPending: isConnecting, isError, connect } = useConnect();
```
Comment (helciofranco) [inline packages/react/src/providers/FuelUIProvider.tsx:62]: are we sure about it?

**Review by helciofranco — APPROVED** [review]

File: packages/react/src/utils/queryKeys.ts [hunk 2]
```diff
@@ -1,18 +1,61 @@
...
-  wallet: 'wallet',
-  transaction: 'transaction',
-  transactionReceipts: 'transactionReceipts',
-  nodeInfo: 'nodeInfo',
-  connectorList: 'connectorList',
-  currentConnector: 'currentConnector',
-  currentNetwork: 'currentNetwork',
+  base: ['fuel'] as QueryKey,
+  account: (): QueryKey => {
+    return [...QUERY_KEYS.base, 'account'];
```
Comment (luizstacio) [inline packages/react/src/utils/queryKeys.ts:7]: we could use also `QUERY_KEYS.base.concat('accounts')` as it creates a new array.

Comment (helciofranco) [reply packages/react/src/utils/queryKeys.ts:7]: nice catch, `concat` is also way faster than the spread operator. I'm fixing it.

Comment (helciofranco) [reply packages/react/src/utils/queryKeys.ts]: https://github.com/FuelLabs/fuels-npm-packs/pull/60/commits/5487d4704e76bb33184ba8bd3e1b424f0b45c3da

**Review by helciofranco — APPROVED** [review]

**Pull request merged**